sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -23,16 +23,17 @@ static Command *getCommand(const EventImplPtr &Event) {
 
 std::vector<EventImplPtr>
 Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
-  Command *Cmd = getCommand(Event);
-  // Command can be nullptr if user creates cl::sycl::event explicitly,
-  // as such event is not mapped to any SYCL task.
-  if (!Cmd)
-    return {};
   std::vector<EventImplPtr> Result;
-  for (const DepDesc &Dep : Cmd->MDeps) {
+
```

Comment:
Suggest calculating number of events we can return here and `reserve` space before pushing back.

Reply:
I reserved space for all the events